### PR TITLE
Feature/additional aws resources

### PR DESCRIPTION
**Pull request by samidbb**
- [x] https://github.com/dfds/backstage/issues/431
- [x] https://github.com/dfds/backstage/issues/495
- [x] https://github.com/dfds/backstage/issues/494
- [x] https://github.com/dfds/backstage/issues/429
- [x] https://github.com/dfds/backstage/issues/430
- [x] https://github.com/dfds/backstage/issues/499
- [x] https://github.com/dfds/backstage/issues/496

### CODE REVIEW

File: package/aws/snstopic/v1alpha1/composition.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      - type: string
+        string:
+          fmt: "%s-aws"
+      policy:
+        fromFieldPath: Required
+  resources:
+  - name: snstopic
+    base:
+      apiVersion: notification.aws.crossplane.io/v1beta1
+      kind: SNSTopic
```
Comment (rifisdfds) [inline package/aws/snstopic/v1alpha1/composition.yaml:24]: Shouldn't the Kind be Topic from the non-deprecated namespace?

File: package/aws/snssubscription/v1alpha1/composition.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    - type: ToCompositeFieldPath
+      fromFieldPath: status.createdResources
+      toFieldPath: status.createdResources.rbac
+      policy:
+        fromFieldPath: Optional    
+
+  - name: snssubscription
+    base:
+      apiVersion: notification.aws.crossplane.io/v1alpha1
+      kind: SNSSubscription
```
Comment (rifisdfds) [inline package/aws/snssubscription/v1alpha1/composition.yaml:47]: Same as with Topic

File: examples/aws/snstopic_not_working/v1alpha1/snstopicclaim.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+apiVersion: messaging.xplane.dfds.cloud/v1alpha1
+kind: XAWSSNSTopicV1alpha1
```
[the remark this answers is not in the document]
Comment (samidbb) [reply examples/aws/snstopic_not_working/v1alpha1/snstopicclaim.yaml:2]: found the issue. it was in composition that was referring to the wrong api version of snstopic

File: package/aws/snssubscription/v1alpha1/composition.yaml (hunk 2 above)
Comment (samidbb) [reply package/aws/snssubscription/v1alpha1/composition.yaml:47]: does it fail to create the claim?

File: package/aws/snstopic/v1alpha1/composition.yaml (hunk 1 above)
Comment (samidbb) [reply package/aws/snstopic/v1alpha1/composition.yaml:24]: you're right! I should have seen that :/ going to update the code

File: package/aws/snssubscription/v1alpha1/composition.yaml (hunk 2 above)
Comment (samidbb) [reply package/aws/snssubscription/v1alpha1/composition.yaml]: done

File: examples/aws/subscription/v1/subscriptionclaim.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+spec:
+  parameters:
+    region: eu-central-1
+    protocol: email
+    # You can see mail sent to this address at the below URL, which is open to.
+    # The public. You'll need to confirm the subscription in order for the
+    # notification to become ready.
+    # https://www.mailinator.com/v4/public/inboxes.jsp?to=crossplane-test
+    endpoint: crossplane-test@mailinator.com
+    topicArn: arn:aws:sns:eu-central-1:069711225475:sample-topic
```
Comment (rifisdfds) [inline examples/aws/subscription/v1/subscriptionclaim.yaml:15]: I think we should blank out any account ID's

**Review by rifisdfds — CHANGES_REQUESTED** [review]
crossplane.yaml still needs amending with the correct minimum required aws-provider

**Review by rifisdfds — APPROVED** [review]

**Pull request merged**